projects/bootloader/src/stm32f0xx/jump_to_bootloader.c

```diff
@@ -0,0 +1,40 @@
+#include "jump_to_bootloader.h"
+
+#include <stdint.h>
+#include <stdnoreturn.h>
+#include <string.h>
+
+#include "bootloader_mcu.h"
+#include "stm32f0xx_misc.h"
+#include "stm32f0xx_syscfg.h"
+
+static noreturn __attribute__((naked)) void prv_perform_jump(uint32_t sp, uint32_t pc) {
+  __asm(
+      "msr msp, %[sp] \n"  // reset the main stack pointer (msp) to sp
+      "bx %[pc] \n"        // jump to pc
+
+      // this bizarre syntax associates the "sp" and "pc" in asm with the sp and pc parameters
+      // see http://www.ethernut.de/en/documents/arm-inline-asm.html
+      :
+      : [sp] "r"(sp), [pc] "r"(pc));
+}
```

Comment:
We might want to factor this function out from this and jump_to_application.c into an ms-common (or ms-bootloader) library called `jump.h` since it just jumps to a particular location.